ubcpi/answer_pool.py

```diff
@@ -0,0 +1,315 @@
+import random
+import persistence as sas_api
+
+POOL_SIZE = 50
+
+
+class UnknownChooseAnswerAlgorithm(Exception):
+    pass
+
+
+def offer_answer(pool, answer, rationale, student_id, algo):
+    """
+    submit a student answer to the answer pool
+
+    The answer maybe selected to stay in the pool depending on the selection algorithm
+
+    Args:
+        pool (dict): answer pool
+            Answer pool format:
+            {
+                option1_index: {
+                    'student_id': { can store algorithm specific info here },
+                    ...
+                }
+                option2_index: ...
+            }
```

Comment:
OK I talked with Dave and he suggests 4k as a reasonable limit. That is kind of complicated to do, so he suggests instead limiting the number of answer choices and the number of responses in the answer pool, so you have a max of N choices x M responses per choice, yielding something around a total max size of 4k. It'd be easy to work through what N and M should be by making example structs, jsonifying them and then looking at the size of the json blob.